client/src/notification/NotificationToaster.js

```diff
@@ -5,21 +5,23 @@
 
 import React, { Fragment, useEffect } from 'react';
 import toast from 'react-hot-toast';
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
+import { CLEAR_NOTIFICATION } from '../actions/types';
 export const NotificationToaster = () => {
   const { error, success } = useSelector((state) => state.notification);
+  const dispatch = useDispatch();
 
   useEffect(() => {
-    if(error){
+    if (error) {
       toast.error(error.data);
     }
-  }, [error])
-
-  useEffect(() => {
-    if(success){
+    if (success) {
```

Comment:
Could there  be a scenario where both these evaluate to true and both success + error toasts show up? I'm thinking if this is possible because of the fact that we are storing both error and success notifications in our store.

Reply:
The only time I can think of it happening is if someone intentionally uses both error and success toasts at the same time, but I'll change it into an else if so that it'll default to the error toast instead of both in that case.
@pmyip3 Feel free to add if you have any thoughts.

Reply:
I agree that it will work fine other than someone intentionally uses both error and success toasts at the same time, as the notification state would be cleared after each rendering of the NotificationToaster because we dispatch a CLEAR_NOTIFICATION action inside the useEffect hook. 
Not quite sure what would happen if we dispatch both error and success message actions at the same time. Cause I don't think it will wait for the completion of the first dispatched action ( as it will go through the state, then the toastNoticaition would get updated because of the changes in state). So we might get both success and error messages existing in the reducer in this situation.
 Another possible solution is we can explicitly clear the error messages in the state when dispatching success messages, vice versa. 
`    case SUCCESS_MESSAGE:
      return {...state, error: null, success: action.payload }`
Then we can ensure only one of the messages would exist in the state at any point of time.